sesman/config.c

```diff
@@ -166,11 +171,45 @@ config_read_globals(int file, struct config_sesman *cf, struct list *param_n,
         cf->enable_user_wm = 0;
     }
 
-    if ('\0' == cf->default_wm[0])
+    if (cf->default_wm == 0)
+    {
+        cf->default_wm = g_strdup("startwm.sh");
+    }
+    else if (g_strlen(cf->default_wm) == 0)
+    {
+        g_free(cf->default_wm);
+        cf->default_wm = g_strdup("startwm.sh");
+    }
+
+    /* if default_wm doesn't begin with '/', it's a relative path from XRDP_CFG_PATH */
+    if (cf->default_wm[0] != '/')
     {
-        g_strncpy(cf->default_wm, "startwm.sh", 11);
+        buf = (char *)g_malloc(1024, 0);
+        g_sprintf(buf, "%s/%s", XRDP_CFG_PATH, g_cfg->default_wm);
```

Comment:
This is prone to buffer overflow

Reply:
Is 4096 enough? https://github.com/torvalds/linux/blob/master/include/uapi/linux/limits.h#L13

Reply:
Will alloc `g_strlen(XRDP_CFG_PATH) + g_strlen(g_cfg->default_wm) + 2`.